_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+    ans = max(ans, x);
+}
+```
+
+N times checking i < N.
+
+N times checking max of ans and x.
+
+N times i++.
+
+N times scanf.
+
+Hence, total number of operations is $$4 * N$$. We only care about dependency on N. Therefore, order is $$O(N)$$.
+
+[CLRS](https://www.google.co.in/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwju8crC0e_UAhWHN48KHTAVCdoQFggjMAA&url=https%3A%2F%2Fweb.njit.edu%2F~wl256%2Fdownload%2Fcs610%2FIntroduction-to-algorithm-3rdEdition.pdf&usg=AFQjCNGJTrGFQdXhlY7ItJk2pCGX9V5S9Q) is having detailed theoretical explanation on Order Analysis.
+Also, for most of the online judges(SPOJ, Codechef etc), number of operations in 1 sec is approximately $$10^8$$.
+In some problems, there are T test cases in one input file. If a solution is having order of $$O(N^3)$$, then total order of the whole input will be $$O(T * N^3)$$.
+
+For Example, if T < 100 and N < $$10^5$$, and order per test case is $$O(N)$$, then number of operations will be at max $$10^5$$ per test case and $$10^7$$ per input file.
+
+## Verdict:
+
+Once you submit your solution, the judge will run a few test cases on your solution and will give you a verdict based on the output of your solution.
+
+**There will be one of following verdicts :**
++   **AC**(Also shown by a green tick): Your solution is accepted.
+
++   **WA**(Also shown by a red cross): Your code gives wrong output. Check for correctness of your approach.
+
++   **TLE**(Also shown by a clock): Your code is taking more time than the given limit. You need to code a faster solution.
+
++   **Run time error**: This could occur due to-
+
+    +   Segmentation fault which could be due to declaring array of size smaller than required or accessing negative indices.
+
+    +   Dividing by zero.
+
+    +   Declaring arrays of size more than $$10^8$$.
+
++   **Compilation Error**: Could be due to wrong language selection or some syntax error. Check the language selected and run it on your PC first and fix syntax errors.
+
+Now, for a warm up, try this problem - [TEST](https://www.codechef.com/problems/TEST).
+
+Few more problems to get you familiar with problem solving--
+
+[https://www.codechef.com/problems/HS08TEST](https://www.codechef.com/problems/HS08TEST)
+
+[https://www.codechef.com/problems/FLOW001](https://www.codechef.com/problems/FLOW001)
+
+[https://www.codechef.com/problems/INTEST](https://www.codechef.com/problems/INTEST)
+
+[https://www.codechef.com/problems/FLOW017](https://www.codechef.com/problems/FLOW017)
+
+---
+
+# Using STL(Standard Template Library)
+
+Certain Data Structures and functions are frequently used while coding solutions. Standard Template Library, commonly known as STL, is basically a collection of few frequently used functions and data structures that can be used directly. This reduces the effort significantly. Also, they are very easy to use and makes the code compact and clean which can be easily debugged.
+
+**Data Structures:**
+
++   Vector
```

Comment:
Tbh, I don't see the point of listing these if we're not going to explain what they are in a line or something.